src/pybind/mgr/dashboard/controllers/host.py

```diff
@@ -254,6 +254,37 @@ def get_inventories(hosts: Optional[List[str]] = None,
     return inventory_hosts
 
 
+def _check_orchestrator_host_op(orch_client, hostname, add_host=True):  # pragma:no cover
+    """Check if we can adding or removing a host with orchestrator
+
+    :param orch_client: Orchestrator client
+    :param add: True for adding host operation, False for removing host
+    :raise DashboardException
+    """
+    host = orch_client.hosts.get(hostname)
+    if add_host and host:
+        raise DashboardException(
+            code='orchestrator_add_existed_host',
+            msg='{} is already in orchestrator'.format(hostname),
+            component='orchestrator')
+    if not add_host and not host:
+        raise DashboardException(
+            code='orchestrator_remove_nonexistent_host',
+            msg='Remove a non-existent host {} from orchestrator'.format(hostname),
+            component='orchestrator')
+
+
+def create_host(hostname: str, addr: Optional[str] = None,
+                labels: Optional[List[str]] = None,
+                maintenance: bool = False):
+    orch_client = OrchClient.instance()
+    _check_orchestrator_host_op(orch_client, hostname, True)
+    orch_client.hosts.add(hostname, addr, labels)
+    if maintenance:
+        orch_client.hosts.enter_maintenance(hostname, False)
+    create_host._cp_config = {'tools.json_in.force': False}  # pylint: disable=W0212
```

Comment:
Don't we have a decorator here? I feel it makes the code messier.

Reply:
> Don't we have a decorator here? I feel it makes the code messier.

Yes there is `allow_empty_body` decorator which does the same job.

Reply:
Thank you. I didn't knew about it. Made the change